torch/_export/utils.py

```diff
@@ -1350,3 +1350,52 @@ def default_flatten_fn_spec(obj, spec) -> list[Any]:
 def deregister_module_as_pytree_input_node(cls: type[torch.nn.Module]) -> None:
     _deregister_pytree_node(cls)
     _deregister_pytree_flatten_spec(cls)
+
+
+def _sync_state(src, dst):
+    assert isinstance(
+        src,
+        torch.nn.Module,
+    ), f"Expected {src} to be a nn.Module"
+    assert isinstance(
+        dst,
+        torch.nn.Module,
+    ), f"Expected {dst} to be a nn.Module"
+    # Share state (params, buffers) between modules.
+    # This ensures that state mutations are visible across them.
+    # Since tensor constants are not mutable, copying (without sharing) is OK.
+    # Also, primitive constants are specialized, so copying (without sharing) is OK.
+    dst._parameters = src._parameters
+    dst._buffers = src._buffers
+
+
+def sync_state(*wrapped_method_modules):
+    """
+    Sync state between exported modules corresponding to wrapped methods.
+    This might be necessary after serializing/deserializing due to copying.
+    """
+    if wrapped_method_modules:
+        m, *other_ms = wrapped_method_modules
+        for other_m in other_ms:
+            _sync_state(m, other_m)
+
+
+class _WrappedMethod(torch.nn.Module):
```

Comment:
I have also seen stuff like this: https://github.com/pytorch/executorch/blob/main/exir/capture/_capture.py#L139 
Just FYI

Reply:
Cool, thanks for the info. Looks like it's temporarily redirecting the original module's forward, seems a bit riskier than wrapping?

Reply:
Should we also delete this: https://github.com/pytorch/pytorch/blob/810d2a3dbd46e7b0fbcca5717b07270336101092/torch/export/_trace.py#L1212

Reply:
Oh lol ok. It seems to be load bearing. I'll investigate separately.